wallet/api/admin_describe_wallet.go

```diff
@@ -0,0 +1,80 @@
+package api
+
+import (
+	"context"
+	"fmt"
+
+	"code.vegaprotocol.io/vega/libs/jsonrpc"
+	"github.com/mitchellh/mapstructure"
+)
+
+type DescribeWalletParams struct {
+	Wallet     string `json:"wallet"`
+	Passphrase string `json:"passphrase"`
+}
+
+type DescribeWalletResult struct {
+	Name    string `json:"name"`
+	ID      string `json:"id"`
+	Type    string `json:"type"`
+	Version uint32 `json:"version"`
+}
+
+type DescribeWallet struct {
+	walletStore WalletStore
+}
+
+// Handle retrieve a wallet from its name and passphrase.
+func (h *DescribeWallet) Handle(ctx context.Context, rawParams jsonrpc.Params) (jsonrpc.Result, *jsonrpc.ErrorDetails) {
+	params, err := validateDescribeWalletParams(rawParams)
+	if err != nil {
+		return nil, invalidParams(err)
+	}
+
+	if exist, err := h.walletStore.WalletExists(ctx, params.Wallet); err != nil {
+		return nil, internalError(fmt.Errorf("couldn't verify wallet existence: %w", err))
+	} else if !exist {
+		return nil, invalidParams(ErrWalletDoesNotExist)
+	}
+
+	w, err := h.walletStore.GetWallet(ctx, params.Wallet, params.Passphrase)
+	if err != nil {
+		return nil, internalError(fmt.Errorf("couldn't retrieve wallet: %w", err))
+	}
+
+	return DescribeWalletResult{
+		Name:    w.Name(),
+		ID:      w.ID(),
+		Type:    w.Type(),
+		Version: w.Version(),
+	}, nil
+}
+
+func validateDescribeWalletParams(rawParams jsonrpc.Params) (DescribeWalletParams, error) {
```

Comment:
Same question here - Is there a specific reason you need the DescribeWalletParams value here?
Isn't it better to return a pointer to it?